src/db.rs

```diff
@@ -1964,7 +1979,7 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_should_recover_imm_from_wal() {
+    async fn test_should_recover_imm_from_wal_after_flush_error() {
```

Comment:
I'm not sure you can do that with failpoints - the `return` action just returns whatever the error that's passed to the macro is. Also in this case I don't think that would really simulate what we want - the memtable flusher would just think that it wrote the sst and then update the manifest to reflect that - so we're really simulating losing an l0 sst, which the implementation doesn't try to recover from.

Here's an alternative proposal:
1. For the first db instance, simulate a fault by configuring failpoints to pause (`fail_parallel::cfg(fp_registry.clone(), "write-compacted-sst-io-error","pause"`). This will just block the memtable flush task (simulating a high-latency memtable flush)
2. Without closing the first db, open the second db without a failpoints registry
3. Verify that we can still read all the data and that the next wal sst id is what we expect
4. Disable the paused failpoint  (`fail_parallel::cfg(fp_registry.clone(), "write-compacted-sst-io-error","off"`) and then close both dbs. The first db will just detect it's been fenced and exit.

Reply:
thanks for the recommendation @rodesai. I'll try to implement that test in that way.

Reply:
I've added the test as suggested by @rodesai

Reply:
> the return action just returns whatever the error that's passed to the macro 

According to the docs, you can do `return(arg)` for the `action`. This will get passed to the closure, and you can then determine what to return using if/else logic or whatever.

> the memtable flusher would just think that it wrote the sst and then update the manifest to reflect that

Yea.. I realized this later. You're right, this won't accomplish what we want. The pause should work, like you said. 👍